Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean

```diff
@@ -226,10 +235,46 @@ instance IsAddHaarMeasure_basis_addHaar (b : Basis ι ℝ E) : IsAddHaarMeasure
   rw [Basis.addHaar]; exact Measure.isAddHaarMeasure_addHaarMeasure _
 #align is_add_haar_measure_basis_add_haar IsAddHaarMeasure_basis_addHaar
 
+instance [TopologicalSpace.SecondCountableTopology E] (b : Basis ι ℝ E) :
+    SigmaFinite b.addHaar := by
+  rw [Basis.addHaar_def]
+  exact MeasureTheory.Measure.sigmaFinite_addHaarMeasure
+
 theorem Basis.addHaar_self (b : Basis ι ℝ E) : b.addHaar (_root_.parallelepiped b) = 1 := by
   rw [Basis.addHaar]; exact addHaarMeasure_self
 #align basis.add_haar_self Basis.addHaar_self
 
+theorem Basis.parallelepiped_measurable [DecidableEq ι] (b : Basis ι ℝ E) :
+    MeasurableSet (b.parallelepiped : Set E) := by
+  have : FiniteDimensional ℝ E := FiniteDimensional.of_fintype_basis b
+  rw [show b.parallelepiped = b.equivFun.toLinearMap ⁻¹'
+      (Set.pi Set.univ fun _ : ι => Set.Icc (0 : ℝ) 1) by
+    ext; simp only [Basis.mem_parallelepiped, Set.mem_Icc, forall_and, LinearEquiv.coe_coe,
+      Pi.le_def, Set.pi_univ_Icc, Set.mem_preimage, equivFun_apply]]
+  refine measurableSet_preimage (LinearMap.continuous_of_finiteDimensional _).measurable ?_
+  exact MeasurableSet.pi Set.countable_univ fun _ _ => measurableSet_Icc
+
+theorem Basis.addHaar_eq  {ι' : Type*} [Fintype ι'] [TopologicalSpace.SecondCountableTopology E]
+    {b : Basis ι ℝ E} {b' : Basis ι' ℝ E} :
+    b.addHaar = b'.addHaar ↔ b.addHaar b'.parallelepiped = 1 :=
```

Comment:
Nice, I think this lemma was the one I probably needed in https://github.com/leanprover-community/mathlib/pull/19013. If you fancy a hopefully-easy follow-up PR, it would be great to have that result (I imagine most of my work there is worthless)

Reply:
> Or I guess I can cherrypick some results from your PR to get this one merge rapidly and then move to #19013?

That sounds good to me; cherry-pick the results into this PR that are both useful for `Complex` and that you think are an improvement over your lemmas, and leave the rest to a later PR.

Reply:
#7037